subprojects/logging/src/main/java/org/gradle/internal/logging/sink/GroupingProgressLogEventGenerator.java

```diff
@@ -226,6 +240,7 @@ synchronized void flushOutput() {
                     listener.onOutput(header());
                 }
 
+                didHaveOutput = bufferedLogs.size() > 0;
```

Comment:
I think it might be easier to follow this if we marked `didHaveOutput` when we buffer?  WDYT?